### PR TITLE
config: kustomize ccruntime deployments

### LINKED ISSUE #144: kustomize ccruntime deployments
**Is your feature request related to a problem? Please describe.**

The `config/samples/` directory holds the deployments for ccruntime. Currently it has two configurations ccruntime and ccruntime-ssh-demo which differs only on their name, payload and runtime classes:

```
# diff ccruntime.yaml ccruntime-ssh-demo.yaml 
4c4
<   name: ccruntime-sample
---
>   name: ccruntime-ssh-demo
13c13
<     payloadImage: quay.io/confidential-containers/runtime-payload:kata-containers-5b7009f2f9cdb39099b97a4f868b53ec9f3c383f
---
>     payloadImage: quay.io/confidential-containers/runtime-payload:kata-containers-ssh-demo-v0.2.0
53,56c53
<     # Uncomment and add the required RuntimeClassNames to be created
<     # If this is commented, then the operator creates 3 default runtimeclasses "kata", "kata-clh", "kata-qemu"
<     runtimeClassNames: 
<       ["kata", "kata-clh", "kata-clh-tdx", "kata-clh-tdx-eaa-kbc", "kata-qemu", "kata-qemu-tdx", "kata-qemu-tdx-eaa-kbc", "kata-qemu-sev"]
---
>     runtimeClassNames: ["kata", "kata-clh", "kata-qemu"]
```

**Describe the solution you'd like**

Use kustomize to manage the ccruntime deployments so that we have ccruntime.yaml as base config and additional overlays for ssh-demo and s390x (to be added in https://github.com/confidential-containers/operator/pull/142).

As an example, on commit ec867f93cc0e38af47239f6bb1384c915c98e476 the enclave-cc runtime was kustomized.

### PR DESCRIPTION
**Pull request by wainersm**
Converted the ccruntime deployments of the default, s390x and ssh-demo variants to be managed by the kustomize tool.

Fixes #144
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

### CODE REVIEW

**Comment by BbolroC** [comment]
Hup, I am on vacation until 8th of Jan. Sorry.

File: config/samples/ccruntime/base/ccruntime.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
       node-role.kubernetes.io/worker: ""
   config:
     installType: bundle
-    payloadImage: quay.io/confidential-containers/runtime-payload:kata-containers-5b7009f2f9cdb39099b97a4f868b53ec9f3c383f
+    payloadImage: quay.io/confidential-containers/runtime-payload:kata-containers
```
Comment (BbolroC) [inline config/samples/ccruntime/base/ccruntime.yaml:13]: I have noticed the tag `kata-containers` does not exist even if it will be updated.

**Comment by BbolroC** [comment]
Other than the small nit above, LGTM. Thanks. 👍 
FYI: Unfortunately, I do not have a permission to approve a PR.

**Comment by wainersm** [comment]
@BbolroC thanks for reviewing it even while in your vacation. Hey, turn off the computer and enjoy it! ;)

**Review by mythi — APPROVED** [review]
/lgtm

**Review by fidencio — APPROVED** [review]
@wainersm, lgtm.

Please, rebase atop of the current main to solve the conflicts and this one would be good to go.

**Comment by fidencio** [comment]
As @wainersm seems to be still on vacation, I've done the quick update in the PR and force-pushed to his branch.

**Pull request merged**